main.py

```
@@ -1,37 +1,91 @@
         try:
             for element in ans:
                 if not (int(element) in interval):
-                    condition = False
+                    good_ans_condition = False
                     break
         except:
-            condition = False
-        if condition:
+            good_ans_condition = False
+        if good_ans_condition:
             if quantity == 1:
                 ans = int(ans[0])
             return ans
         else:
             print("Something went wrong...")
 
 
-def init_matrix(lines):
-    matrix = []
-    columns = -1
-    for i in range(1,lines+1):
-        line = get_input("Line " + str(i) + ": ", columns, range(MIN_NUMBER, MAX_NUMBER))
-        if i==1:
-            columns = len(line)
-        matrix.append(line)
-    return matrix, lines, columns
+def get_option():  # Print all the options and get user's response
+    print("Options: \n", options)
+    return get_input("Enter: ", 1, range(1, len(options) + 1))
+
+
+class Matrix:
+    def __init__(self):
+        # Create the Matrix size n*m:
+        self.matrix, self.n, self.m = self.get_matrix()
+
+    def init_matrix(self, lines):  # Initialization the matrix from the keyboard
+        matrix = []
+        columns = -1
+        for i in range(1, lines + 1):
+            line = get_input("Line " + str(i) + ": ", columns, range(MIN_NUMBER, MAX_NUMBER + 1))
+            if i == 1:
+                columns = len(line)
+            matrix.append(line)
+        return matrix, lines, columns
+
+    def get_matrix(self):  # Find out the number of lines in the matrix n*m.
+        self.matrix, n, m = self.init_matrix(
+            get_input("Enter the number of lines of the matrix n (1-99): ", 1, range(1, 100)))
+        return self.matrix, n, m
+
+    def det(self, matrix, n, m):  # Count determinant
+        if n != m:
+            print("The determinant of the matrix can be found only in a square matrix.")
```

Minor: better to throw an exception